src/Tools/dotnet-gcdump/DotNetHeapDump/MemoryGraph.cs

```diff
@@ -36,8 +37,28 @@ public void WriteAsBinaryFile(string outputFileName)
         }
         public static MemoryGraph ReadFromBinaryFile(string inputFileName)
         {
-            Deserializer deserializer = new(inputFileName);
-            deserializer.TypeResolver = typeName => System.Type.GetType(typeName);  // resolve types in this assembly (and mscorlib)
+            Deserializer deserializer = new(inputFileName, SerializationSettings.Default);
```

Comment:
@brianrob Let me know if this seems reasonable? I did not see a way to use the Default factory directly.

Reply:
This effectively gets you back to the previous behavior where any type that is encoded into the serialized file can be created on-demand.  This capability was removed from TraceEvent during the security review, and so you will not want this to remain this way.  Instead, you'll need to specify the set of expected types so that unexpected types cause a failure at deserialization time.

From a review of this class in TraceEvent, it looks like you may be able to get away without any of this: https://github.com/microsoft/perfview/blob/main/src/MemoryGraph/MemoryGraph.cs#L34.

Depending on timeline and ability to test, you can decide if you should fix this immediately, or file an issue and come back to it.  I just wouldn't let this linger for long since it is a known security vulnerability.

Reply:
(Sorry for the delay, I didn't get a notification of this for some reason.)

Reply:
This seems to be working as you suggested. Thanks!